### PR TITLE
Packages: Import photon package

### PR DESCRIPTION
**Pull request by sirreal**
#### Changes proposed in this Pull Request

* Pulls in photon source
* ~Preps a 3.0.0 version~
* Update to match Calypso style (es6, Jest…)

#### Testing instructions

* Tests pass?
* Shrinkwrap changes look correct?
* Photon works correctly in Calypso (reader, theme showcase…)

### CODE REVIEW

File: packages/photon/Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
```
[the remark this answers is not in the document]
Comment (sirreal) [reply packages/photon/Makefile]: 👍

File: packages/photon/History.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+3.0.0 / 2018-12-05
```
Comment (sirreal) [inline packages/photon/History.md]: Update before release…

File: packages/photon/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/** @format */
+
+module.exports = {
+	extends: '../../.eslintrc.js',
```
Comment (sirreal) [inline packages/photon/.eslintrc.js]: Not needed.

File: packages/photon/LICENSE [hunk 4]
```diff
@@ -0,0 +1,24 @@
+(The MIT License)
```
Comment (sirreal) [inline packages/photon/LICENSE]: MIT? Does this need to be GPL or GPLv2?

Remove the license and use the package.json property

**Comment by sirreal** [comment]
Rebased with a few changes I'd noted and a fresh `update-deps`. Ready for review.

**Comment by sirreal** [comment]
Rebased with a few changes I'd noted and a fresh `update-deps`. Ready for review.

It looks like the photon may need some work for #29559, I'd like to get this landed.

**Comment by sirreal** [comment]
Desktop tests are failing, seem to be exposing a problem with the package or with the desktop setup. We'll need to investigate…

**Comment by blowery** [comment]
@adlk any ideas on the failing desktop build here?
@sirreal I think we fixed some issues with the desktop build after this PR was created. Maybe we just need to drop the branch with the same name from the desktop repo and let it get re-created before the test run?

**Comment by blowery** [comment]
(just did the branch dance with desktop. we'll see)

**Comment by blowery** [comment]
new branch rebuild didn't help :/

**Comment by sirreal** [comment]
I've rebased and removed the desktop test branch, let's see if it's happy now.

**Comment by sirreal** [comment]
This seems to be working just fine in the themes showcase where it's used:

https://hash-36207191d5e75c4537ae2e55c118912e2e879160.calypso.live/themes

However e2e tests are failing, seemingly unrelated.

**Comment by bsessions85** [comment]
e2e tests failures are unrelated. We've been having some issues I am working on fixing right now.  This should be good

**Comment by sirreal** [comment]
Needs to be updated after photon 3 release with https://github.com/Automattic/photon.js/pull/10

File: .eslintignore [hunk 5]
```diff
@@ -3,7 +3,7 @@
 /build/
 /docs/
 /public/
-/packages/*/dist/**/*
```
Comment (blowery) [inline .eslintignore]: can drop this change i think

File: packages/photon/package-lock.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (blowery) [inline packages/photon/package-lock.json]: can drop the package lock entirely and add a .npmrc that disallows package locks

File: packages/photon/package.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{
+	"name": "photon",
+	"version": "2.0.1",
+	"description": "JavaScript library for the WordPress.com Photon image manipulation service",
+	"main": "dist/cjs/index.js",
+	"module": "dist/esm/index.js",
```
Comment (blowery) [inline packages/photon/package.json]: should add `sideEffects: false`

Comment (sirreal) [reply packages/photon/package.json]: 4f03cbd4743db2f8f83a6745d6d9adff634df87e

File: packages/photon/package-lock.json (hunk 6 above)
Comment (sirreal) [reply packages/photon/package-lock.json]: 28b76fb770b98ac19fc500a54acdc78cd136835c

File: packages/photon/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (sirreal) [reply packages/photon/Makefile]: bfad93e4bb212d8021b7eeaea091abed624db591

File: .eslintignore (hunk 5 above)
Comment (sirreal) [reply .eslintignore]: Dropped the change. Should be synonymous: ignore the dir, or ignore everything in all the dirs in the dir 🙂

**Comment by sirreal** [comment]
Oh hey, the desktop tests passed! Let's move ahead with this 🤞

**Comment by blowery** [comment]
@sirreal what do you think, do we want to update calypso to use it too? Or just import it for this PR?

**Comment by sirreal** [comment]
Let's use it, I'll update this branch and give it a test 👍

**Comment by sirreal** [comment]
One note: 

With

- https://github.com/Automattic/photon.js/pull/10
- https://github.com/Automattic/photon.js/pull/11

Photon repo is ahead of this repo. However, that published version was deprecated, the latest tag was rolled back, and the enhancement may be stalled indefinitely:

- https://github.com/Automattic/photon.js/pull/12
- pMz3w-9qA-p2

I'm not sure whether it's worth pulling those changes into this branch or what. That will be a concern for the next release because it may need to be v4. It would be nice to get conclusion to pMz3w-9qA-p2.

**Comment by sirreal** [comment]
- Added `file:` ref to use the package.
- Updated deps `npm run update-deps`
- Tested reader, themes, media editor, photon seems to be working well 👍

**Review by blowery — APPROVED** [review]
This looks good!

**Comment by blowery** [comment]
Looks like the package.json just switched from tabs to spaces (or vice versa)

**Comment by ockham** [comment]
Seems like this broke jurassic.ninja's Jetpack blocks build step, since that didn't run `npm ci && npm run build-packages` but was still using `lerna bootstrap --ci`.

Found by @jeherve and fixed by @oskosk. Convo: p1550870061189900-slack-jetpack-gutenberg

**Pull request merged**